### DISCUSSION TITLE
Unable to save tiff as Ccittfax4

### DISCUSSION THREAD

**Discussion by flavien-lory** [libvips/libvips#2731] [Q&A]
**Unable to save tiff as Ccittfax4**

**To Reproduce**

```C#
... Loading image... 
var gs = img.Colourspace(Enums.Interpretation.Bw);
var bw = gs < 128;
// CASE 1
bw.Tiffsave($"./Out/example_{page}.tif",Enums.ForeignTiffCompression.Ccittfax4,null,null,null,null,null,null,null,null,
                    1,null,null,null,null ,null,null, null , null, null,null,null,null,null);
// CASE 2
bw.Tiffsave($"./Out/example_{page}.tif",Enums.ForeignTiffCompression.Ccittfax4,null,null,null,null,null,null,null,null,
                    null,null,null,null,null ,null,null, null , null, null,null,null,null,null);
```

**Actual behavior**

CASE 1 tiffsave does not support optional argument: bitdepth  // Work fine on win64 with native image
CASE 2 Fax3SetupState: Bits/sample must be 1 for Group 3/4 encoding/decoding      // Without BitDepth parameter

**Environment**

 - OS: linux docker env. libpoppler82 libvips42
 - Vips: NetVips version 8.7.4 
[LibImg.zip](https://github.com/libvips/libvips/files/8316636/LibImg.zip)

**Comment by jcupitt** [comment]
Hi @flavien-lory,

You are using a very old libvips, from before the `bitdepth` parameter was added. I would look at updating to the current version. For example, you can probably use:

https://www.nuget.org/packages/NetVips.Native.linux-x64/

In 8.7, I think you need the `squash` parameter (ie. squash 8 bit images down to 1 bit).

You don't need all those nulls. Try:

```C#
(gs < 128).WriteToFile($"./Out/example_{page}.tif",  new VOption()
{
    { "compression", ForeignTiffCompression.Ccittfax4 },
    { "squash", true },
});
```

`squash` still works in current libvips.
